pkg/comp-functions/functions/helper/funcio.go

```diff
@@ -0,0 +1,236 @@
+package helper
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"sort"
+	"strings"
+
+	xkube "github.com/crossplane-contrib/provider-kubernetes/apis/object/v1alpha1"
+	"github.com/pmezard/go-difflib/difflib"
+	"github.com/vshn/appcat/pkg"
+	"github.com/vshn/appcat/pkg/comp-functions/runtime"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	k8sruntime "k8s.io/apimachinery/pkg/runtime"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// SaveFuncIOToConfigMap is a helper function to make it easy to determine the contents of the FunctionIO.
+// This function is not a normall composition function! It won't add anything to the functionIO object to
+// avoid it being bloated.
+// It will keep two configmaps:
+// previous generation functionIO
+// current generation functionIO
+// The config maps are always written to the default namespace and contain the
+// composite name as the name.
+func SaveFuncIOToConfigMap(funcName string) func(context.Context, *runtime.Runtime) runtime.Result {
+	return func(ctx context.Context, iof *runtime.Runtime) runtime.Result {
+
+		kubeClient, err := client.New(ctrl.GetConfigOrDie(), client.Options{
+			Scheme: pkg.SetupScheme(),
+		})
+		if err != nil {
+			return runtime.NewFatalErr(ctx, "cannot create kube client", err)
+		}
```

Comment:
Why do you create a new kube client in every reconciliation. Wouldn't it be smarter to pass it as an argument? That would also make it testable